auditlogger.go

```diff
@@ -0,0 +1,58 @@
+package ladon
+
+import (
+	"log"
+	"os"
+	"strings"
+)
+
+// AuditLogger tracks denied and granted authorizations.
+type AuditLogger interface {
+	LogRejectedAccessRequest(request *Request, pool Policies, deciders Policies)
+	LogGrantedAccessRequest(request *Request, pool Policies, deciders Policies)
+}
+
+// AuditLoggerNoOp is the default AuditLogger, that tracks nothing.
+type AuditLoggerNoOp struct{}
+
+func (*AuditLoggerNoOp) LogRejectedAccessRequest(r *Request, p Policies, d Policies) {}
+func (*AuditLoggerNoOp) LogGrantedAccessRequest(r *Request, p Policies, d Policies)  {}
+
+var DefaultAuditLogger = &AuditLoggerNoOp{}
+
+// AuditLoggerInfo outputs information about granting or rejecting policies.
+type AuditLoggerInfo struct {
+	Logger *log.Logger
+}
+
+func (a *AuditLoggerInfo) logger() *log.Logger {
+	if a.Logger == nil {
+		a.Logger = log.New(os.Stderr, "", log.LstdFlags)
+	}
+	return a.Logger
+}
+
+func (a *AuditLoggerInfo) LogRejectedAccessRequest(r *Request, p Policies, d Policies) {
+	if len(d) > 1 {
+		allowed := joinPoliciesNames(d[0 : len(d)-1])
+		denied := d[len(d)-1].GetID()
+		a.Logger.Printf("policies %s allow access, but policy %s forcefully denied it", allowed, denied)
```

Comment:
Start with upper case, use log level `Infof`

Reply:
This is the stdlib logger, not logrus
https://golang.org/pkg/log/#Logger.Printf

Reply:
Right, my bad!